src/CommunicationExecutor.c

```diff
@@ -207,6 +207,62 @@ void Ros_Communication_PublishActionFeedback(rcl_timer_t* timer, int64_t last_ca
     Ros_ActionServer_FJT_ProcessResult();
 }
 
+void Ros_Communication_MonitorUserLanState(SHORT interface_id)
+{
+    //assume link is currently up (and pretend it was up before this task even
+    //started).
+    //This task is only started once MotoROS2 can reach the Agent (and can
+    //connect to it), so the link MUST be up or otherwise this task could
+    //not have been started.
+    //If we start with the assumption the link is down, we'd see a state change
+    //as soon as we call Ros_UserLan_IsLinkUp(..), which would not lead to any
+    //action (as we only act on up->down edges), but would be unnecessary.
+    BOOL bLinkWasUp = TRUE;
+
+    //we set g_Ros_Communication_AgentIsConnected=FALSE in the body below,
+    //so this exits when a link drop is detected
+    while (g_Ros_Communication_AgentIsConnected)
+    {
+        //TODO(gavanderhoorn): make configurable ?
+        Ros_Sleep(0.5 * 1e3);
+
+        //retrieve current state of monitored link, assume it's down
+        BOOL bLinkIsUp = FALSE;
+        STATUS status = Ros_UserLan_IsLinkUp(interface_id, &bLinkIsUp);
+        if (status != OK)
+        {
+            //TODO(gavanderhoorn): should this be fatal for MotoROS2?
```

Comment:
Should it perhaps be?

If a user has kept the link state monitor enabled (by not disabling it via the config file), it might be reasonable to say any problem trying to run it should probably cause alarms to be raised.

Reply:
But that doesn't post any alarms, does it?

Reply:
You're correct there's no alarm. I'd be ok with adding an alarm, but I still don't think it should be fatal. The user can try a different cable w/o rebooting.

Reply:
An alarm will now be posted. Non-fatal.